spec/factories/rents.rb

```diff
@@ -2,8 +2,8 @@
 
 FactoryBot.define do
   factory :rent do
-    user { user_id }
-    book { book_id }
+    user_id { create(:user).id }
```

Comment:
You don't need to specify id and do `user { create(:user) }`, so its a bit clearer.